paparazzi/src/main/java/app/cash/paparazzi/Paparazzi.kt

```diff
@@ -272,12 +280,21 @@ class Paparazzi(
       val constructor =
         bridgeSessionClass.getDeclaredConstructor(RenderSessionImpl::class.java, Result::class.java)
       constructor.isAccessible = true
-      return constructor.newInstance(renderSession, result) as BridgeRenderSession
+      return (constructor.newInstance(renderSession, result) as BridgeRenderSession).apply {
+        val viewGroup = rootViews[0].viewObject as View
+        val viewRootImpl = viewGroup.viewRootImpl
+        val params = sessionParamsBuilder.build()
+        viewRootImpl.setView(viewGroup.findRoot(), LayoutParams(params.screenWidth, params.screenHeight, 0, 0, 0), null)
+      }
     } catch (e: Exception) {
       throw RuntimeException(e)
     }
   }
 
+  private fun View.findRoot(): View {
```

Comment:
super nit: +`tailrec`

You may have to pull the base case into its own conditional separate from the recursion case though

Reply:
👍  Nice, ended up changing the compiled code into a while loop with no recursion